client/src/components/UserSearch.tsx

```diff
@@ -32,6 +32,7 @@ export function UserSearch(props: UserProps) {
 
   const handleSearch = async (value: string) => {
     if (value) {
+      value = value.trim();
```

Comment:
I suppose it makes sense to validate trimmed value inside `if`, ie do trim beforehand

Reply:
I did not understand at what point it is necessary to make a trim? Do you mean before the function (onSearch={val => handleSearch(val.trim())}) or before the if block? Something like this?                             
```
value = value.trim();                                       
 if (value) {
   const data = await searchFn(value, onlyStudentsWithoutMentorShown);
   ...
}
```

Reply:
like in your example above. the thing is that if value=` `, it will pass the `if` condition, but due to trim you will call search with empty value

Reply:
fix this